azure-batch/azure/batch/custom/patch.py

```diff
@@ -0,0 +1,285 @@
+                self._raw)
+        except BatchErrorException as e:
+            # In case of a chunk exceeding the MaxMessageSize split chunk in half
+            # and resubmit smaller chunk requests
+            # TODO: Replace string with constant variable once available in SDK
+            if e.error.code == "RequestBodyTooLarge":  # pylint: disable=no-member
+                # In this case the task is misbehaved and will not be able to be added due to:
+                #   1) The task exceeding the max message size
+                #   2) A single cell of the task exceeds the per-cell limit, or
+                #   3) Sum of all cells exceeds max row limit
+                if len(chunk_tasks_to_add) == 1:
+                    failed_task = chunk_tasks_to_add.pop()
+                    results_queue.appendleft(e)
+                    logging.error("Task ID " + failed_task.id + " failed to add due to exceeding the request body" +
+                                    " being too large")
+                    self._has_early_termination_error = True
+                else:
+                    # Assumption: Tasks are relatively close in size therefore if one batch exceeds size limit
+                    # we should decrease the initial task collection size to avoid repeating the error
+                    # Midpoint is lower bounded by 1 due to above base case
+                    midpoint = int(len(chunk_tasks_to_add) / 2)
+                    # Restrict one thread at a time to do this compare and set,
+                    # therefore forcing max_tasks_per_request to be strictly decreasing
+                    with self._max_tasks_lock:
+                        if midpoint < self._max_tasks_per_request:
+                            self._max_tasks_per_request = midpoint
+                            logging.info("Amount of tasks per request reduced from " +
+                                            str(self._max_tasks_per_request) + " to " + str(midpoint) +
+                                            " due to the request body being too large")
+
+                    # Not the most efficient solution for all cases, but the goal of this is to handle this
+                    # exception and have it work in all cases where tasks are well behaved
+                    # Behavior retries as a smaller chunk and
+                    # appends extra tasks to queue to be picked up by another thread .
+                    self._tasks_to_add.extendleft(chunk_tasks_to_add[midpoint:])
+                    self._bulk_add_tasks(results_queue, chunk_tasks_to_add[:midpoint])
+            elif 500 <= e.response.status_code <= 599:
+                self._tasks_to_add.extendleft(chunk_tasks_to_add)
+            else:
+                results_queue.appendleft(e)
+        except Exception as e:  # pylint: disable=broad-except
+            results_queue.appendleft(e)
+        else:
+            if isinstance(add_collection_response, ClientRawResponse):
```

Comment:
Do EAFP
https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python

```python
try:
    add_collection_response = add_collection_response.output
except AttributeError:
    pass
```
unless add_collection_response has a possiblity to have "ouput" (which I don't think so)